### PR TITLE
Revert "Prune unzipped source distributions in `uv cache prune --ci` (#7446)"

### LINKED ISSUE #7485: uv 0.4.11 failing to read distribution cache in github actions
Our Github Actions CI started failing with the following error when running `uv pip install pyspark==3.2.1`

```
Using Python 3.8.18 interpreter at: /__w/_tool/Python/3.8.18/x64/bin/python3
Creating virtualenv with seed packages at: venv
warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
 + pip==24.2
 + setuptools==75.1.0
 + wheel==0.44.0
error: Failed to download and build `pyspark==3.2.1`
  Caused by: Failed to read from the distribution cache
  Caused by: failed to read directory `/__w/_temp/setup-uv-cache/built-wheels-v3/pypi/pyspark/3.2.1/IStJ-q5m0o-G-eruFFpEK/pyspark-3.2.1.tar.gz`
  Caused by: No such file or directory (os error 2)
```

This started happening after the latest 0.4.11 release.

<details><summary>Workflow Declaration</summary>

```yaml
- name: Setup Python
  uses: actions/setup-python@v5
  with:
    python-version: '3.8'

- name: Install uv
  uses: astral-sh/setup-uv@v2
  with:
    enable-cache: true

- run: |
    uv venv venv --seed
    uv pip install -r requirements.txt pyarrow pyspark==3.2.1 --python venv  # <--- FAILS HERE
  shell: bash
```

</details>

<details><summary><code>astral-sh/setup-uv</code> logs</summary>

```
> Run astral-sh/setup-uv@v2
  with:
    enable-cache: true
    version: latest
    github-token: ***
  env:
    pythonLocation: /__w/_tool/Python/3.8.18/x64
    PKG_CONFIG_PATH: /__w/_tool/Python/3.8.18/x64/lib/pkgconfig
    Python_ROOT_DIR: /__w/_tool/Python/3.8.18/x64
    Python2_ROOT_DIR: /__w/_tool/Python/3.8.18/x64
    Python3_ROOT_DIR: /__w/_tool/Python/3.8.18/x64
    LD_LIBRARY_PATH: /__w/_tool/Python/3.8.18/x64/lib:/opt/oracle/instantclient_21_8/
/usr/bin/docker exec  b1f1b19e402573bba171b6a6d208b411385eed518d4f720827a31a93cb1a6f88 sh -c "cat /etc/*release | grep ^ID"
Downloading uv from "https://github.com/astral-sh/uv/releases/latest/download/uv-x86_64-unknown-linux-gnu.tar.gz" ...
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /__w/_temp/5138b5d3-39fd-4adf-abe8-9a6b348864e2 -f /__w/_temp/ef8159c7-bfe3-439b-b565-b6a11cf5126c
Added /__w/_tool/uv/0.4.11/x86_64 to the path
Successfully installed uv version latest
Set UV_CACHE_DIR to /__w/_temp/setup-uv-cache
Trying to restore uv cache from GitHub Actions cache with key: setup-uv-1-x86_64-unknown-linux-gnu-0.4.11
-no-dependency-glob
Received 0 of 301099792 (0.0%), 0.0 MBs/sec
Received 33554432 of 301099792 (11.1%), 16.0 MBs/sec
Received 67108864 of 301099792 (22.3%), 21.3 MBs/sec
Received 109051904 of 301099792 (36.2%), 26.0 MBs/sec
Received 150994944 of 301099792 (50.1%), 28.8 MBs/sec
Received 184549376 of 301099792 (61.3%), 29.3 MBs/sec
Received 213909504 of 301099792 (71.0%), 29.1 MBs/sec
Received 251658240 of 301099792 (83.6%), 30.0 MBs/sec
Received 288516880 of 301099792 (95.8%), 30.5 MBs/sec
Cache Size: ~287 MB (301099792 B)
/usr/bin/tar -xf /__w/_temp/92d12651-008f-4643-8d55-d87145b39532/cache.tgz -P -C /__w/<PRIVATE-REPOSITORY> -z
Received 301099792 of 301099792 (100.0%), 28.7 MBs/sec
Cache restored successfully
uv cache restored from GitHub Actions cache with key: setup-uv-1-x86_64-unknown-linux-gnu-0.4.11
```
</details>

<details><summary><code>uv pip install ...</code> complete logs</summary>

```
> Run uv venv venv --seed
  uv venv venv --seed
  uv pip install -r requirements.txt pyarrow pyspark==3.2.1 --python venv
  shell: bash --noprofile --norc -e -o pipefail {0}
  env:
    pythonLocation: /__w/_tool/Python/3.8.18/x64
    PKG_CONFIG_PATH: /__w/_tool/Python/3.8.18/x64/lib/pkgconfig
    Python_ROOT_DIR: /__w/_tool/Python/3.8.18/x64
    Python2_ROOT_DIR: /__w/_tool/Python/3.8.18/x64
    Python3_ROOT_DIR: /__w/_tool/Python/3.8.18/x64
    LD_LIBRARY_PATH: /__w/_tool/Python/3.8.18/x64/lib:/opt/oracle/instantclient_21_8/
    UV_CACHE_DIR: /__w/_temp/setup-uv-cache
    JAVA_HOME: /__w/_tool/Java_Corretto_jdk/8.422.05-1/x64
    JAVA_HOME_8_X64: /__w/_tool/Java_Corretto_jdk/8.422.05-1/x64
    PNPM_HOME: /github/home/setup-pnpm/node_modules/.bin
Using Python 3.8.18 interpreter at: /__w/_tool/Python/3.8.18/x64/bin/python3
Creating virtualenv with seed packages at: venv
warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
 + pip==24.2
 + setuptools==75.1.0
 + wheel==0.44.0
error: Failed to download and build `pyspark==3.2.1`
  Caused by: Failed to read from the distribution cache
  Caused by: failed to read directory `/__w/_temp/setup-uv-cache/built-wheels-v3/pypi/pyspark/3.2.1/IStJ-q5m0o-G-eruFFpEK/pyspark-3.2.1.tar.gz`
  Caused by: No such file or directory (os error 2)
```

</details>

Tried reproducing this on my local machine with `uv self update` followed by a `uv pip install pyspark==3.2.1` but couldn't reproduce the failure.

CI Runner Info
---

1. All our actions run inside our private CI docker image which is based on Ubuntu 22.04 LTS
2. Host OS
   ```
   Distribution: Amazon Linux release 2023.5.20240805 (Amazon Linux)
   Architecture: x86_64
   ```

### PR DESCRIPTION
**Pull request by BurntSushi**
This reverts commit 9f7d9da449cab4119d07da3ecd2e3b868e092afd from PR #7446.

This is meant to be available as a quick fix if we can't otherwise find
a quick fix to the problem reported in #7485.

Fixes #7485

### CODE REVIEW

File: crates/uv-cache/src/lib.rs [hunk 1]
```diff
@@ -740,8 +740,7 @@ pub enum CacheBucket {
 impl CacheBucket {
     fn to_str(self) -> &'static str {
         match self {
-            // Note, next time we change the version we should change the name of this bucket to `source-dists-v0`
-            Self::SourceDistributions => "built-wheels-v3",
+            Self::SourceDistributions => "source-dists-v0",
```
Comment (zanieb) [inline crates/uv-cache/src/lib.rs:743]: Woo haha

**Comment by zanieb** [comment]
@BurntSushi can you test with #7498 too?

**Comment by BurntSushi** [comment]
@zanieb Yup, #7498 resolves it too.

**Comment by BurntSushi** [comment]
Closing in favor of #7498.

**Pull request closed without merging**